### PR TITLE
Feature/specify letters conversion

### PR DESCRIPTION
**Pull request by cukejianya**
### ******* **DO NOT MERGE** *******

There is still a lot to do. But I implemented the function. I also refactor the `convert` function in order to keep the code DRY and implement the 3 bulletin (**If both are present, then don't convert letters that don't much any regex**) 

### TO DO:
* [x]  Add **upper** and **lower** options and implement them.
* [x]  Comment options using jsdoc
* [x]  Add unit test for **upper** and **lower** options
* [x]  Add declaration of **upper** and **lower** options for Typescript, [Declaration Files](https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html)
* [x]  Add usage and describe **upper** and **lower** options in README
* [ ] Squash Commits

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -145,11 +157,33 @@ function randomCase(input) {
...
 
+/**
+ * Converts matched letters to a specified case
+ * @param {string} input - string to convert
+ * @param {(string | RegExp)} regex - use to match letters you want to convert
+ * @caseType {string} caseType - the case type
+ * @return the given input with match letters to the  case type determined
+ */
+function convertToSpecificCase(input, regex, caseType) {
+  if (caseType !== "lower" && caseType !== "upper") {
```
Comment (strdr4605) [inline index.js]: I think is more clear if `str => str.toUpperCase()` or `str => str.toLowerCase()` will be passed as the 3rd argument. And named like `replaceCallback` or [`replacer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_function_as_a_parameter) and there is no need to check for argument but just `return input.replace(regex, replaceCallback);`

File: index.js [hunk 2]
```diff
@@ -145,11 +157,33 @@ function randomCase(input) {
   return convert(input, () => Math.round(Math.random()));
 }
 
+/**
+ * Converts matched letters to a specified case
+ * @param {string} input - string to convert
+ * @param {(string | RegExp)} regex - use to match letters you want to convert
+ * @caseType {string} caseType - the case type
```
Comment (strdr4605) [inline index.js]: There is a typo:
```suggestion
 * @param ...
```

File: mockingcase.test.js [hunk 3]
```diff
@@ -200,4 +201,184 @@ describe("mockingcase", () => {
       expect(consoleOutput).toEqual(expectedOutput);
     });
   });
+
+  describe(`option.upper | set as String`, () => {
+    const lowerCaseInput = "hello world";
+    const capitalizeInput = "Hello World";
+    const upperCaseInput = "HELLO WORLD";
+
+    test("it should make match substring upper case", () => {
```
Comment (strdr4605) [inline mockingcase.test.js]: `it` is not necessary here as tests can be written also with `it("...`. To not make changes everywhere will stick with `test("...` but in my opinion `it` is redundant here.
```suggestion
    test("Should make match substring upper case", () => {
```

File: mockingcase.test.js [hunk 4]
```diff
@@ -200,4 +201,184 @@ describe("mockingcase", () => {
       expect(consoleOutput).toEqual(expectedOutput);
     });
   });
+
+  describe(`option.upper | set as String`, () => {
```
Comment (strdr4605) [inline mockingcase.test.js]: Can you also cover with tests **Error throws **?

**Comment by strdr4605** [comment]
@cukejianya, I added eslint to the project, can you please rebase and resolve conflicts?!

**Comment by cukejianya** [comment]
@strdr4605 Sorry for the delay. I was busy working on other things this week. I should be able to get the readme done tonight - if not - tomorrow at the latest.

Also, can you double check the declaration file. Not sure if I did that right.

File: index.d.ts [hunk 5]
```diff
@@ -7,6 +7,8 @@ declare module "mockingcase" {
     random?: boolean;
     onlyLetters?: boolean;
     firstUpper?: boolean;
+    upper?: string | object;
```
Comment (strdr4605) [inline index.d.ts]:
```suggestion
    upper?: string | RegExp;
```

Comment (strdr4605) [reply index.d.ts]: > Also, can you double check the declaration file. Not sure if I did that right.

I made the changes from my comment  and successfully tested in a typescript project

**Pull request merged**